server/src/main/java/io/crate/execution/jobs/Task.java

```diff
@@ -21,12 +21,24 @@
 
 package io.crate.execution.jobs;
 
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
 import io.crate.concurrent.CompletionListenable;
 import io.crate.data.Killable;
 
 public interface Task extends CompletionListenable<Void>, Killable {
 
-    void start();
+    /**
+     * Start a task.
+     * @return future that is completed once the start action completed its preparation phase.
+     *         This does *NOT* indicate that the task finished. Use {@link #completionFuture()} for that.
+     *         If null the start method is synchronous and users can assume the task is initiated once
+     *         the method returns.
```

Comment:
What is the benefit of `@Nullable` here, like why not always expect a future? Wouldn't this make things more complicated as the return value must be checked?

Reply:
To avoid creating and chaining futures per task in https://github.com/crate/crate/blob/6613911f1199c4325d7d6d8c5a1a68c1bbcda86c/server/src/main/java/io/crate/execution/jobs/RootTask.java#L188-L210 

We've many queries where there is never a fetch